header.go

```diff
@@ -0,0 +1,59 @@
+package xinvoice
+
+import (
+	"time"
+
+	"github.com/invopop/gobl/bill"
+	"github.com/invopop/gobl/cal"
+	"github.com/invopop/gobl/cbc"
+)
+
+// ExchangedDocument a collection of data for a Cross Industry Invoice Header that is exchanged between two or more parties in written, printed or electronic form.
+type Header struct {
+	ID           string `xml:"ram:ID"`
+	TypeCode     string `xml:"ram:TypeCode"`
+	IssueDate    *Date  `xml:"ram:IssueDateTime>udt:DateTimeString"`
+	IncludedNote *Note  `xml:"ram:IncludedNote,omitempty"`
+}
+
+// Note defines note in the RAM structure
+type Note struct {
+	Content     string `xml:"ram:Content"`
+	SubjectCode string `xml:"ram:SubjectCode"`
+}
+
+func NewHeader(inv *bill.Invoice) *ExchangedDocument {
+	id := inv.Series + "-" + inv.Code
+	typeCode := invoiceTypeCode(inv.Type)
+	date := formatIssueDate(inv.IssueDate)
+
+	return &ExchangedDocument{
+		ID:       id,
+		TypeCode: typeCode,
+		IssueDate: &Date{
+			Date:   date,
+			Format: "102",
+		},
+	}
+}
+
+func formatIssueDate(date cal.Date) string {
+	t := time.Date(date.Year, date.Month, date.Day, 0, 0, 0, 0, time.UTC)
+	return t.Format("20060102")
+}
+
+// For German suppliers, the element "Invoice type code" (BT-3) should only contain the
+// following values from code list UNTDID 1001:
+// - 326 (Partial invoice)
+// - 380 (Commercial invoice)
+// - 384 (Corrected invoice)
+// - 389 (Self-billed invoice)
```

Comment:
Not essential here, but GOBL supports self-billed invoices using tags: https://github.com/invopop/gobl/blob/main/regimes/common/invoice_tags.go#L40-L51

Reply:
The type for self-billed invoice is in the tax package and for the rest it is in the bill package won't there be a problem with that? I'm not sure how those interact.

Reply:
They're independent. Tags are added inside the invoice in the `tax` and `tags` property, whereas the type is set in `type`. We decided in GOBL to only support a basic set of concise invoice types, then apply tags to cover exceptions.

We think this makes sense, as there is no reason why we should prevent somebody from self-billing themselves a credit note or corrective invoice for example.

Reply:
Makes sense 👍. I'll keep that in mind, but we will probably handle this case after we are done with credit notes.